### ISSUE TITLE
Route53 API authorization in EC2 instances

### ISSUE THREAD

**Issue by madhusudancs** [kubernetes/kubernetes#27467]
Either verify that the EC2 VMs that are part of the cluster federation can access Route53 APIs or define the necessary IAM policies to allow these VMs to access Route53 APIs.

Equivalent authorization scopes were added to GCE and GKE in #27357 and #27466

cc @nikhiljindal @quinton-hoole @kubernetes/sig-cluster-federation

**Comment by madhusudancs** [comment]
@justinsb assigning it you. I hope you can help us with this. But please feel free to reassign as necessary.

**Comment by nikhiljindal** [comment]
Adding to 1.3 milestone

**Comment by erictune** [comment]
Let's find another assignee.  @quinton-hoole

**Comment by erictune** [comment]
Justin has several things he is already working on for 1.3 and this seems like something someone else could tackle.

**Comment by justinsb** [comment]
Sorry this slipped off my plate

We just need to add this to the IAM policy/policies `cluster/aws/templates/iam/kubernetes-master-policy.json` and `cluster/aws/templates/iam/kubernetes-minion-policy.json`:

```
    {
      "Effect": "Allow",
      "Action": ["route53:*"],
      "Resource": ["*"]
    },
```

Question though - just on the masters, or on master & nodes?

**Comment by quinton-hoole** [comment]
Thanks @justinsb.  The Federation control plane pods run on nodes, so the answer to your question is "on nodes".

**Comment by quinton-hoole** [comment]
More specifically it runs in pods on nodes.

**Comment by justinsb** [comment]
Fix in #27794 .  I did it on the master & on the nodes; I figure the nodes are the ones we are more concerned about security on, and not having it on the master is just going to cause confusion.

**Comment by justinsb** [comment]
Also, I just do it across the board, not just in federation.  I know a lot of users are excited to have k8s auto-provision DNS with ingress (and get LetsEncrypt working)...
